axelar-chains-config/src/utils/verifyContract.js

```diff
@@ -1,5 +1,139 @@
 const { execSync } = require('child_process');
 const { writeFileSync } = require('fs');
+const fs = require('fs');
+const path = require('path');
+const axios = require('axios');
+
+async function getSmallestFile(directoryPath) {
+    const fileNames = await fs.promises.readdir(directoryPath);
+    let smallestFile = null;
+    let smallestSize = Number.MAX_SAFE_INTEGER;
+
+    for (const fileName of fileNames) {
+        const filePath = path.join(directoryPath, fileName);
+        const stats = await fs.promises.stat(filePath);
+
+        if (stats.size < smallestSize) {
+            smallestFile = filePath;
+            smallestSize = stats.size;
+        }
+    }
+
+    return smallestFile;
+}
+
+function getContractFileName(config, address) {
+    for (const [key, currentValue] of Object.entries(config)) {
+        if (currentValue === address) {
+            return key.charAt(0).toUpperCase() + key.slice(1);
+        }
+    }
+
+    return null;
+}
+
+async function findFilePath(startPath, targetFileName) {
+    if (!fs.existsSync(startPath)) {
+        throw new Error(`Start path does not exist: ${startPath}`);
+    }
+
+    const directoriesToSearch = [startPath];
+
+    while (directoriesToSearch.length > 0) {
+        const currentDirectory = directoriesToSearch.pop();
+        const filesAndDirectories = fs.readdirSync(currentDirectory, { withFileTypes: true });
+
+        for (const fileOrDirectory of filesAndDirectories) {
+            const fullPath = path.join(currentDirectory, fileOrDirectory.name);
+
+            if (fileOrDirectory.isDirectory()) {
+                directoriesToSearch.push(fullPath);
+            } else if (fileOrDirectory.isFile() && fileOrDirectory.name === targetFileName) {
+                return fullPath;
+            }
+        }
+    }
+
+    throw new Error(`File not found: ${targetFileName}`);
+}
+
+async function verifyFilfox(env, contract, options) {
+    const { dir, contractName } = options;
+
+    if (!dir || !contractName) {
+        throw new Error('Invalid verification options');
+    }
+
+    const config = require(`../../info/${env}.json`);
+    const contractConfig = config.chains.filecoin.contracts[contractName];
+    const contractFileName = getContractFileName(contractConfig, contract);
```

Comment:
isn't contractName already being passed in? you can enforce it to be passed in, and update usages in our code instead of the limited reverse map

Reply:
The config map has different names for addresses than actual contract file names, so this should fail @deanamiel.

Reply:
@milapsheth , @deanamiel can we modify all `verifyContract()` calls within`verify-contract.js` to have a hardcoded contract name with it, which will simplify script execution for all the contracts. I will also be able to use this in my PR: https://github.com/axelarnetwork/axelar-contract-deployments/pull/162

Reply:
The `contractName` being passed would just be InterchainTokenService whereas I need each individual contract name within ITS to find the correct contract path to the flattened source code. @blockchainguyy yeah that is true, it will not work for a few contract names, I thought of adding an optional argument for the hardcoded contract name to `verifyContract()` as well but am not sure it's worth all these changes for only filfox verification for their specific verification endpoint requirements, thoughts @milapsheth ?